### DISCUSSION TITLE
请问怎样才能适配MySQL8，我们不想使用plsql，求大神指条明路

### DISCUSSION THREAD

**Discussion by wenlicheng** [goharbor/harbor#20837] [Q&A] [answered]
# Harbor 配置文件

# 用于访问管理 UI 和注册服务的 IP 地址或主机名
hostname: 192.168.47.130

# HTTP 相关配置
http:
  # HTTP 端口，默认为 80。启用 HTTPS 时，此端口将重定向到 HTTPS 端口
  port: 10001

# HTTPS 相关配置（未启用）
# https:
#   port: 443
#   certificate: /your/certificate/path
#   private_key: /your/private/key/path

# IP 协议栈配置（默认为仅启用 IPv4）
# ip_family:
#   ipv6:
#     enabled: false
#   ipv4:
#     enabled: true

# 内部 TLS 通信配置（未启用）
# internal_tls:
#   enabled: true
#   dir: /etc/harbor/tls/internal

# 外部代理配置（未启用）
# external_url: https://reg.mydomain.com:8433

# Harbor 管理员初始密码
harbor_admin_password: Harbor112421

# Harbor 数据库配置
database:
  # Harbor 使用的数据库类型，默认是 postgresql，你需要修改为 mysql
  provider: mysql
  # 数据库的主机地址（外部MySQL服务器的IP地址）
  host: 192.168.47.130
  # 数据库端口号（通常是3306）
  port: 3306
  # 数据库用户名
  username: harbor
  # 数据库密码
  password: harbor111
  # 数据库名称（Harbor使用的数据库名称，确保在外部MySQL中已创建）
  dbname: harbor
  
 # 其他MySQL连接参数
 #这个参数用于设置MySQL连接的SSL模式。`disable` 表示禁用SSL连接，使用普通的未加密连接。通常在测试环境或内部网络中使用。
  sslmode: disable
  
  #这个参数设置数据库连接池中最大的空闲连接数。`100` 表示在空闲状态下最多保留100个连接，以提高性能和减少连接创建的开销。
  max_idle_conns: 100

  #这个参数设置数据库连接池中最大的打开连接数。`900` 表示最多允许900个连接同时打开，控制数据库并发连接的数量。
  max_open_conns: 900

  #这个参数设置连接在连接池中的最大生命周期。`5m` 表示每个连接在池中最多可以存在5分钟，超时后将被关闭，确保连接的健康状态和释放老旧连接。
  conn_max_lifetime: 5m
  
  #这个参数设置连接在连接池中最大的空闲时间。`0` 表示不限制连接在空闲状态下的最大时间，连接将保持空闲直到被关闭或替换。如果设置为非零值，连接在空闲超过指定时间后将被关闭。
  conn_max_idle_time: 0

# 默认数据卷
data_volume: /data/harbor

# Harbor 存储配置（默认使用本地文件系统）
# storage_service:
#   filesystem:
#     maxthreads: 100
#   redirect:
#     disable: false

# Trivy 配置
trivy:
  ignore_unfixed: false
  skip_update: false
  skip_java_db_update: false
  offline_scan: false
  security_check: vuln
  insecure: false
  timeout: 5m0s

# Job 服务配置
jobservice:
  max_job_workers: 10
  job_loggers:
    - STD_OUTPUT
    - FILE
  logger_sweeper_duration: 1 # 天

# 通知配置
notification:
  webhook_job_max_retry: 3
  webhook_job_http_client_timeout: 3 # 秒

# 日志配置
log:
  level: info
  local:
    rotate_count: 50
    rotate_size: 200M
    location: /var/log/harbor

# 外部 syslog 端点配置（未启用）
# external_endpoint:
#   protocol: tcp
#   host: localhost
#   port: 5140

# Redis 配置（使用外部 Redis 时请取消注释）
# external_redis:
#   host: redis:6379
#   password: 
#   registry_db_index: 1
#   jobservice_db_index: 2
#   trivy_db_index: 5
#   idle_timeout_seconds: 30

# 全局代理配置
proxy:
  http_proxy:
  https_proxy:
  no_proxy:
  components:
    - core
    - jobservice
    - trivy

# 指标配置（未启用）
# metric:
#   enabled: false
#   port: 9090
#   path: /metrics

# Trace 相关配置（未启用）
# trace:
#   enabled: true
#   sample_rate: 1

# 启用 purge _upload 目录
upload_purging:
  enabled: true
  age: 168h
  interval: 24h
  dryrun: false

# 缓存层配置
cache:
  enabled: false
  expire_hours: 24

# Harbor 核心配置（未启用）
# core:
#   quota_update_provider: redis

这个是我的配置文件

**Comment by stonezdj** [comment, marked as answer]
Harbor doesn't support MySQL

**Reply by wenlicheng** [reply]
大佬，怎样修可以使用呢，或者有先例嘛？
拜托拜托
